jwst/jump/tests/test_detect_jumps.py

```diff
@@ -31,17 +60,312 @@ def test_onecr_10_groups(setup_inputs):
 
-def test_onecr_10_groups_fullarray(setup_inputs):
+def test_twoints_onecr_each_10_groups_neighbors_flagged(setup_inputs):
+    """"
+        Two integrations with CRs in different locations. This makes sure we are correctly
+        dealing with integrations.
+    """
+    grouptime = 3.0
+    ingain = 200
+    inreadnoise = np.float64(7)
+    ngroups = 10
+    model1, gdq, rnModel, pixdq, err, gain = setup_inputs(ngroups=ngroups, nints=2,
+                                                          gain=ingain, readnoise=inreadnoise, deltatime=grouptime)
+    # two segments perfect fit, second segment has twice the slope
+    model1.data[0, 0, 5, 5] = 15.0
+    model1.data[0, 1, 5, 5] = 20.0
+    model1.data[0, 2, 5, 5] = 25.0
+    model1.data[0, 3, 5, 5] = 30.0
+    model1.data[0, 4, 5, 5] = 35.0
+    model1.data[0, 5, 5, 5] = 140.0
+    model1.data[0, 6, 5, 5] = 150.0
+    model1.data[0, 7, 5, 5] = 160.0
+    model1.data[0, 8, 5, 5] = 170.0
+    model1.data[0, 9, 5, 5] = 180.0
+    model1.data[1, 0, 15, 5] = 15.0
+    model1.data[1, 1, 15, 5] = 20.0
+    model1.data[1, 2, 15, 5] = 25.0
+    model1.data[1, 3, 15, 5] = 30.0
+    model1.data[1, 4, 15, 5] = 35.0
+    model1.data[1, 5, 15, 5] = 40.0
+    model1.data[1, 6, 15, 5] = 50.0
+    model1.data[1, 7, 15, 5] = 160.0
+    model1.data[1, 8, 15, 5] = 170.0
+    model1.data[1, 9, 15, 5] = 180.0
+    out_model = detect_jumps(model1, gain, rnModel, 4.0, False, 4.0, 1, 200, 4, True)
+    assert (4 == np.max(out_model.groupdq[0, 5, 5, 5]))
+    assert (4 == out_model.groupdq[0, 5, 5, 6])
+    assert (4 == out_model.groupdq[0, 5, 5, 4])
+    assert (4 == out_model.groupdq[0, 5, 6, 5])
+    assert (4 == out_model.groupdq[0, 5, 4, 5])
+    assert (4 == out_model.groupdq[1, 7, 15, 5])
+    assert (4 == out_model.groupdq[1, 7, 15, 6])
+    assert (4 == out_model.groupdq[1, 7, 15, 4])
+    assert (4 == out_model.groupdq[1, 7, 16, 5])
+    assert (4 == out_model.groupdq[1, 7, 14, 5])
+
+def test_flagging_of_CRs_across_slice_boundaries(setup_inputs):
+    """"
+            A multiprocesssing test that has two CRs on the boundary between two
+            slices. This makes sure that we are correctly flagging neighbors in different
+            slices.
+    """
     grouptime = 3.0
-    ingain = 5  # use large gain to show that Poisson noise doesn't affect the recombination
+    ingain = 200
+    inreadnoise = np.float64(7)
+    ngroups = 10
+
+    model1, gdq, rnModel, pixdq, err, gain = setup_inputs(ngroups=ngroups, nints=2,
+                                                          gain=ingain, readnoise=inreadnoise,
+                                                          deltatime=grouptime)
+    nrows = model1.data.shape[3]
+    num_cores = multiprocessing.cpu_count()
+    max_cores = 'half'
+    numslices = num_cores // 2
+    yincrement = int(nrows / numslices)
+    # two segments perfect fit, second segment has twice the slope
+    #add a CR on the last row of the first slice
```

Comment:
There was an error in this test that if the number of slices is 1, it would write outside the array. I added a test to skip the test if the number of slices is 1. Otherwise it works with any number of slices greater than 1.